synapse/handlers/device.py

```diff
@@ -484,15 +490,18 @@ async def notify_device_update(
 
         room_ids = await self.store.get_rooms_for_user(user_id)
 
-        hosts: Set[str] = set()
-        if self.hs.is_mine_id(user_id):
-            for room_id in room_ids:
-                joined_users = await self.store.get_users_in_room(room_id)
-                hosts.update(get_domain_from_id(u) for u in joined_users)
+        hosts: Optional[Set[str]] = None
+        if self.use_new_device_lists_changes_in_room:
```

Comment:
is this the wrong way round? we should *skip* calculating hosts if `use_new_device_lists_changes_in_room`?

Reply:
Bah, yes. And this is the only place that uses it directly ─ everything else infers it from `hosts` being `None` ─ so nothing broke. 

I'm not sure if there is an easy way to test that this is the right way round?

Reply:
you might be able to block the `_handle_new_device_update_async` job in the tests (eg by setting `_handle_new_device_update_is_processing`), and check what actually gets written to the db?

Not sure it's worth it though.

Reply:
I guess this code is going to be ripped out in the next release anyway